be/src/olap/generic_iterators.cpp

```diff
@@ -262,6 +266,9 @@ Status MergeIterator::next_batch(RowBlockV2* block) {
         // copy current row to block
         copy_row(&dst_row, ctx->current_row(), block->pool());
 
+        if (ctx->is_partial_delete()) {
+            block->set_delete_state(DEL_PARTIAL_SATISFIED);
```

Comment:
I can not understand why you set block DEL_PARTIAL_SATISFIED, I think you will saturate the block. All rows of this block are meaningful.

Reply:
Here the block is not filtered yet, should should be set delete status.
Delete can be calculated here, but need to refactor condition to realize it. which will be done in another pr.